tso-server/src/main/java/org/apache/omid/tso/PersistenceProcessorHandler.java

```diff
@@ -39,75 +39,82 @@
                                 ReplyProcessor replyProcessor,
-                                RetryProcessor retryProc,
+                                RetryProcessor retryProcessor,
                                 Panicker panicker)
     throws InterruptedException, ExecutionException, IOException {
 
         this.tsoHostAndPort = tsoHostAndPort;
         this.leaseManager = leaseManager;
         this.writer = commitTable.getWriter();
         this.replyProcessor = replyProcessor;
-        this.retryProc = retryProc;
+        this.retryProcessor = retryProcessor;
         this.panicker = panicker;
 
-        flushTimer = metrics.timer(name("tso", "persist", "flush"));
-        batchSizeHistogram = metrics.histogram(name("tso", "persist", "batchsize"));
+        // Metrics in this component
+        flushTimer = metrics.timer(name("tso", "persist", "flush", "latency"));
+        flushedCommitEventsHistogram = metrics.histogram(name("tso", "persist", "flushed", "commits", "size"));
+        batchSizeHistogram = metrics.histogram(name("tso", "persist", "batch", "size"));
 
     }
 
     @Override
-    public void onEvent(PersistenceProcessorImpl.PersistBatchEvent event) throws Exception {
+    public void onEvent(PersistenceProcessorImpl.PersistBatchEvent batchEvent) throws Exception {
 
-        Batch batch = event.getBatch();
-        for (int i=0; i < batch.getNumEvents(); ++i) {
-            PersistEvent localEvent = batch.get(i);
+        int commitEventsToFlush = 0;
+        Batch batch = batchEvent.getBatch();
+        int numOfBatchedEvents = batch.getNumEvents();
+        batchSizeHistogram.update(numOfBatchedEvents);
+        for (int i=0; i < numOfBatchedEvents; ++i) {
+            PersistEvent event = batch.get(i);
 
-            switch (localEvent.getType()) {
+            switch (event.getType()) {
             case COMMIT:
-                localEvent.getMonCtx().timerStart("commitPersistProcessor");
+                event.getMonCtx().timerStart("commitPersistProcessor");
                 // TODO: What happens when the IOException is thrown?
-                writer.addCommittedTransaction(localEvent.getStartTimestamp(), localEvent.getCommitTimestamp());
+                writer.addCommittedTransaction(event.getStartTimestamp(), event.getCommitTimestamp());
+                commitEventsToFlush++;
                 break;
             case ABORT:
                 break;
             case TIMESTAMP:
-                localEvent.getMonCtx().timerStart("timestampPersistProcessor");
+                event.getMonCtx().timerStart("timestampPersistProcessor");
                 break;
-            default:
-                throw new RuntimeException("Unknown event type: " + localEvent.getType().name());
             }
         }
-        if (batch.getNumEvents() > 0) {
-            flush(batch.getNumEvents());
-            sendReplies(batch, event.getBatchSequence());
+
+        if(commitEventsToFlush > 0) {
```

Comment:
Done. Now it reads like this:

 void flush(int commitEventsToFlush) {

        commitSuicideIfNotMaster();
        try {
            long startFlushTimeInNs = System.nanoTime();
            if(commitEventsToFlush > 0) {
                writer.flush();
            }
            flushTimer.update(System.nanoTime() - startFlushTimeInNs);
            flushedCommitEventsHistogram.update(commitEventsToFlush);
        } catch (IOException e) {
            panicker.panic("Error persisting commit batch", e);
        }
        commitSuicideIfNotMaster();

    }